pwa/src/services/validateJSON.ts

```diff
@@ -1,12 +1,12 @@
-export const validateJSON = (_json: string): boolean => {
+export const validateJSON = (_json: string): boolean|string => {
   const RegExp: RegExp = /^[0-9]+$/;
 
-  if (RegExp.test(_json)) return false;
+  if (RegExp.test(_json)) return 'Invalid JSON';
 
   try {
     JSON.parse(_json);
     return true;
   } catch {
-    return false;
+    return 'Invalid JSON';
   }
 };
```

Comment:
Couple of things here:

1. I think we can expand a bit on the error message;
2. We're not duplicating the error message;
3. We have to validate when there's no data sent (due to the way react-hook-form works) and lastly
4. We've named the prop `_json`, but we're actually validating some data to check **if** it is JSON, so it's not accurately named right now.

I propose the following:


```
export const validateJSON = (data: string): boolean | string => {
  const RegExp: RegExp = /^[0-9]+$/;
  const invalidMessage = "This field contains invalid JSON";

  if (!data) return true; // empty data should not be validated

  if (RegExp.test(data)) return invalidMessage;

  try {
    return JSON.parse(data) && true;
  } catch {
    return invalidMessage;
  }
};
```